client/state/data-layer/wpcom/concierge/shifts/index.js

```diff
@@ -0,0 +1,54 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { noRetry } from 'state/data-layer/wpcom-http/pipeline/retry-on-failure/policies';
+import { updateConciergeShifts } from 'state/concierge/actions';
+import { errorNotice } from 'state/notices/actions';
+import { CONCIERGE_SHIFTS_FETCH } from 'state/action-types';
+
+export const requestFetchConciergeShifts = ( { dispatch }, action ) => {
+	const { scheduleId } = action;
+
+	dispatch(
+		http(
+			{
+				method: 'GET',
+				path: `/concierge/schedules/${ scheduleId }/shifts`,
+				apiNamespace: 'wpcom/v2',
+				retryPolicy: noRetry(),
+			},
+			action
+		)
+	);
+};
+
+export const storeFetchedConciergeShifts = ( { dispatch }, action, shifts ) =>
+	dispatch( updateConciergeShifts( shifts ) );
+
+export const showConciergeShiftsFetchError = ( { dispatch } ) =>
+	dispatch(
+		errorNotice(
+			translate(
+				"We've encountered problems trying to load available shifts. Please try again later."
+			)
+		)
+	);
+
+export default {
+	[ CONCIERGE_SHIFTS_FETCH ]: [
```

Comment:
Just a note here somewhat mentioned before, but our nomenclature for the data layer code stresses `_REQUEST` action suffixes over `_FETCH` to distinguish between actions which actually force something to happen and those which indicate a data need which Calypso will try to fulfill.

In other words, there's a tendency to see a 1:1 correlation between `_FETCH` action dispatches and network requests, but no such relationship between `_REQUEST` action dispatches and network requests. `_REQUEST` is saying, "I need this data," while `_FETCH` communicates, "Get this data now"

Reply:
Ah, right. That's something that I missed. I've updated the naming in b7ce14b and 1675197.